prism/src/main/java/org/apache/falcon/service/BacklogMetricEmitterService.java

```diff
@@ -100,6 +100,41 @@ protected SimpleDateFormat initialValue() {
     private static ConcurrentHashMap<Entity, List<MetricInfo>> entityBacklogs = new ConcurrentHashMap<>();
 
     @Override
+    public void onAdd(Entity entity) throws FalconException{
+        //DO Nothing
+    }
+
+    @Override
+    public void onRemove(Entity entity) throws FalconException{
+        if (entity.getEntityType() != EntityType.PROCESS){
+            return;
+        }
+        synchronized (this){
```

Comment:
Should be sufficient to synchronize on entityBacklogs, right? Same applies for onChange method.